From 4b8af1a07e59062de872749dfdaa693184413270 Mon Sep 17 00:00:00 2001
From: Thomas Guillem <thomas@gllm.fr>
Date: Thu, 21 Apr 2022 08:20:21 +0200
Subject: [PATCH 79/80] smb2: fix lseek return type

It was causing seek error when seeking past INT_MAX

Regression from c23709134c77dc70d66cc3420ceadfe08b28ed44
---
 modules/access/smb2.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/modules/access/smb2.c b/modules/access/smb2.c
index a8e71ad528..3065d10120 100644
--- a/modules/access/smb2.c
+++ b/modules/access/smb2.c
@@ -315,7 +315,7 @@ FileSeek(stream_t *access, uint64_t i_pos)
 
     struct vlc_smb2_op op = VLC_SMB2_OP(access, &sys->smb2);
 
-    int err = smb2_lseek(op.smb2, sys->smb2fh, i_pos, SEEK_SET, NULL);
+    int64_t err = smb2_lseek(op.smb2, sys->smb2fh, i_pos, SEEK_SET, NULL);
     if (err < 0)
     {
         VLC_SMB2_SET_ERROR(&op, "smb2_lseek", err);
-- 
2.32.0 (Apple Git-132)

